### PR TITLE
refactor: break down the design page(the first step)

### PR DESCRIPTION
**Pull request by lei9444**
## Description
**In this PR**

This is the first step to break down the design page. This PR just split the design page to four parts, but the states in every component is not optimal. 
image

Create this PR to avoid too many conflicts

**Next step**
Optimize every component and states

## Task Item
refs #5559

### CODE REVIEW

File: Composer/packages/client/src/pages/design/Listener.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        focused,
+        promptTab: getTabFromFragment(),
+      });
+
+      /* eslint-disable no-underscore-dangle */
+      // @ts-ignore
+      globalHistory._onTransitionComplete();
+      /* eslint-enable */
+    }
+  }, [location]);
```
Comment (GeoffCoxMSFT) [inline Composer/packages/client/src/pages/design/Listener.tsx:84]: Does this need more dependencies like the things this function uses from props?

File: Composer/packages/client/src/recoilModel/selectors/dialogs.ts [hunk 2]
```diff
@@ -25,3 +25,14 @@ export const dialogsSelectorFamily = selectorFamily<DialogInfo[], string>({
...
 });
+
+export const currentDialogState = selectorFamily<DialogInfo, { projectId: string; dialogId?: string }>({
+  key: 'currentDialog',
+  get: ({ projectId, dialogId }) => ({ get }) => {
+    if (dialogId) {
+      return get(dialogState({ projectId, dialogId }));
+    }
+
+    return get(dialogsSelectorFamily(projectId))[0];
```
Comment (srinaath) [inline Composer/packages/client/src/recoilModel/selectors/dialogs.ts:36]: get(dialogsSelectorFamily(projectId))?.\[0\]. Null reference check

File: Composer/packages/client/src/pages/design/CommandBar.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    const selectedActions = visualEditorSelection.map((id) => get(currentDialog?.content, id));
+    const showDisableBtn = selectedActions.some((x) => get(x, 'disabled') !== true);
+    const showEnableBtn = selectedActions.some((x) => get(x, 'disabled') === true);
+
+    return { actionSelected, showDisableBtn, showEnableBtn };
+  }, [visualEditorSelection, currentDialog?.content]);
+
+  const EditorAPI = getEditorAPI();
+
+  const toolbarItems: IToolbarItem[] = [
```
Comment (srinaath) [inline Composer/packages/client/src/pages/design/CommandBar.tsx:50]: Can this be memoized?

Comment (srinaath) [reply Composer/packages/client/src/pages/design/CommandBar.tsx:50]: Or do u think its better to get this PR in to get the structure and we can optimize in successive PR's

File: Composer/packages/client/src/pages/design/Listener.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { dispatcherState, currentDialogState } from '../../recoilModel';
+import { decodeDesignerPathToArrayPath } from '../../utils/convertUtils/designerPathEncoder';
+
+const getTabFromFragment = () => {
+  const tab = window.location.hash.substring(1);
+
+  return Object.values(PromptTab).find((value) => tab === value);
+};
+
+const Listener: React.FC<RouteComponentProps<{ dialogId: string; projectId: string; skillId?: string }>> = (props) => {
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/Listener.tsx:20]: Please pick a better name for this or at least add comments describing the responsibilities of this faceless component.

Even better, consider creating a hook.

File: Composer/packages/client/src/pages/design/PropertyPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  const { isRemote: isRemoteSkill } = useRecoilValue(projectMetaDataState(projectId));
+  const skillsByProjectId = useRecoilValue(skillNameIdentifierByProjectIdSelector);
+  const skills = useRecoilValue(skillsStateSelector);
+
+  const skillManifestFile = useMemo(() => {
+    if (!isSkill) return undefined;
+
+    const skillNameIdentifier = skillsByProjectId[projectId];
+    return skills[skillNameIdentifier];
+  }, [skills, isSkill]);
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/PropertyPanel.tsx:42]: skillsByProjectId dep

File: Composer/packages/client/src/pages/design/SideBar.tsx [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+        onBotStart={startSingleBot}
+        onBotStop={stopSingleBot}
+        onDialogCreateTrigger={(projectId, dialogId) => {
+          setTriggerModalInfo({ projectId, dialogId });
+        }}
+        onDialogDeleteTrigger={handleDeleteTrigger}
+        onErrorClick={handleErrorClick}
+        onSelect={handleSelect}
+      />
+      <Modals projectId={projectId} />
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/SideBar.tsx:268]: I think this belongs to DesignPage not SideBar, what do you think?

**Review by hatpick — CHANGES_REQUESTED** [review]

**Comment by lei9444** [comment]
Thanks @GeoffCoxMSFT  @srinaath @hatpick 
Do you think we can merge this PR first if all checks pass? Then I can do optimize for every single component in successive PR's

**Comment by srinaath** [comment]
I think it makes sense to get this PR in early. The more we delay more workstreams are going to refactor their code furthur later.

**Review by hatpick — APPROVED** [review]

**Review by srinaath — APPROVED** [review]

**Pull request merged**